code/modules/power/supermatter/supermatter_process.dm

```diff
@@ -54,20 +54,25 @@
 			env.merge(removed)
 			air_update_turf(FALSE, FALSE)
 	else
-		if(takes_damage)
-			//causing damage
-			deal_damage(removed)
-
-		//registers the current enviromental gases in the various lists and vars
-		setup_lists(removed)
-		//some gases can have special interactions
-		special_gases_interactions(env, removed)
-		//main power calculations proc
-		power_calculations(env, removed)
-		//irradiate at this point
-		emit_radiation()
-		//handles temperature increase and gases made by the crystal
-		temperature_gas_production(env, removed)
+
+		if(!hypermatter_state)
+			if(takes_damage)
+				//causing damage
+				deal_damage(removed)
+
+			//registers the current enviromental gases in the various lists and vars
+			setup_lists(removed)
+			//some gases can have special interactions
+			special_gases_interactions(env, removed)
+			//main power calculations proc
+			power_calculations(env, removed)
+			//irradiate at this point
+			emit_radiation()
+			//handles temperature increase and gases made by the crystal
+			temperature_gas_production(env, removed)
+
+		else
+			handle_hypermatter_state()
```

Comment:
this is really hhhhhhh but I'm not sure how to fix it

Reply:
yeah i know, but unless i add ```if(hypermatter_state)
return``` to all these procs i don't see another way to handle it